src/compiler.c

```diff
@@ -3735,7 +3739,7 @@ static void CompIf(Stat stat)
     /* print a comment                                                     */
     if ( CompPass == 2 ) {
         Emit( "\n/* if " );
-        PrintExpr(READ_EXPR(stat, 0));
+        PrintExpr(BODY_FUNC(CURR_FUNC()), READ_EXPR(stat, 0));
```

Comment:
I think this is fine because `READ_EXPR` reads from `BODY_FUNC(CURR_FUNC())`, and it is very explicit here that this is precisely the body we pass to `PrintExpr`... inside `PrintExpr`, though, no use is made of `BODY_FUNC(CURR_FUNC())` or `STATE(PtrBody)` or anything.

Of course next the compiler code should be made to not use these globals -- that was partially done in the commits I removed from this PR, but it wasn't quite complete.